### PR TITLE
Memberships: Add cache to get_post_access_level

### PR DESCRIPTION
**Pull request by lezama**
Fixes p1712662633338699-slack-C052XEUUBL4

## Proposed changes:

* Add cache to get_post_access_level improving performance at least for post with lots of comments

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Comment by millerf** [comment]
It tests fine. I was not expecting our meta cache to not being able to handle so many comments.

I just find one small caveat: the cache is not aware of the blog ID, so it could potentially return the wrong access level if a blog could have the same post ID. It would then get pretty tricky to debug...

We could add the `blog ID` in the cache key, but we recently ran into issue with this approach: some jobs run through a lot of blogs and would get out of memory ( PR here: D143925-code ).

We need to bust the cache on `switch_to_blog` event.

**Comment by lezama** [comment]
thanks for the review @millerf, good call on adding the blog_id to the key, added in https://github.com/Automattic/jetpack/pull/36798/commits/6555eff3bf7d1b48e9da96d1c98de17d224a7dc9

**Review by millerf — CHANGES_REQUESTED** [review]
Using cache-objet is an issue as the cache is not static. IT is also the cache implemented in `get_post_meta` so we know it will not solve the issue.

As already commented earlier, we need to bust the cache on `switch_to_blog`.

File: projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php [hunk 1]
```diff
@@ -539,10 +554,20 @@ public static function get_post_access_level( $post_id = null ) {
 			return Abstract_Token_Subscription_Service::POST_ACCESS_LEVEL_EVERYBODY;
 		}
 
+		$blog_id   = get_current_blog_id();
+		$cache_key = $blog_id . '_' . $post_id;
```
Comment (millerf) [inline projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php:558]: This is unnecessary now

**Review by millerf — APPROVED** [review]
WFM and looks fine.
Removing the `blog id` from the key now could be a nice finishing touch. But I leave it to you...

**Comment by lezama** [comment]
> Removing the blog id from the key now could be a nice finishing touch. But I leave it to you...

let's leave it for now, it will remind us to think about multisite

**Pull request merged**